crates/jose/src/jwk/public_parameters.rs

```diff
@@ -113,27 +105,24 @@ impl ParametersInfo for RsaPublicParameters {
 }
 
 impl RsaPublicParameters {
-    pub const fn new(n: Vec<u8>, e: Vec<u8>) -> Self {
+    pub const fn new(n: Base64UrlNoPad, e: Base64UrlNoPad) -> Self {
         Self { n, e }
     }
 }
 
-#[serde_as]
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, JsonSchema)]
 pub struct EcPublicParameters {
     pub(crate) crv: JsonWebKeyEcEllipticCurve,
 
     #[schemars(with = "String")]
-    #[serde_as(as = "Base64<UrlSafe, Unpadded>")]
-    x: Vec<u8>,
+    x: Base64UrlNoPad,
 
     #[schemars(with = "String")]
-    #[serde_as(as = "Base64<UrlSafe, Unpadded>")]
-    y: Vec<u8>,
+    y: Base64UrlNoPad,
 }
 
 impl EcPublicParameters {
-    pub const fn new(crv: JsonWebKeyEcEllipticCurve, x: Vec<u8>, y: Vec<u8>) -> Self {
+    pub const fn new(crv: JsonWebKeyEcEllipticCurve, x: Base64UrlNoPad, y: Base64UrlNoPad) -> Self {
```

Comment:
I think I would prefer to keep the bytes parameters as `Vec<u8>` since the base64 is only a serialization/deserialization detail? It does mean that `Base64::new` should be `const` though.

This is not a strong opinion, if it really was more convenient to have them like that, it's fine with me